### ISSUE TITLE
arrow-return-shorthand incorrect

### ISSUE THREAD

**Issue by MrAndersen1** [palantir/tslint#2223]
### Bug Report

- __TSLint version__: `4.4.2`
- __TypeScript version__: `2.3.0-dev.20170220`
- __Running TSLint via__: `Visual Studio Code as well as directly on the command-line`

#### TypeScript code being linted

```ts
interface testInterface {
    "something": number;
}
[].map((e): testInterface => { return { "something": 1 }; });
```

#### Actual behavior

'This arrow function body can be simplified by omitting the curly braces and the keyword 'return'. (arrow-return-shorthand)'

#### Expected behavior

no warning as it cannot be simplified by omitting the curly bracers and the keyword 'return'.

**Comment by MrAndersen1** [comment]
Upon reading more about it, however, it turns out you can replace the return and bracers with regular parenthesis ` => ({ "something": 1 })` so only the warning message would need updating for this case.

**Comment by DethAriel** [comment]
Given the comment by @MrAndersen1 I think the warning can be enhanced to handle this type of return statement and suggest to also wrap the thing into brackets

**Comment by icfantv** [comment]
I have to say, though, that wrapping the thing in parenthesis is not intuitive and doesn't look right - even though it may be.

**Comment by MrAndersen1** [comment]
I believe this has been fixed even though it says here that just 3 of 4 tasks are complete.
